test/common/buffer/buffer_speed_test.cc

```diff
@@ -247,11 +248,12 @@ BENCHMARK(BufferReserveCommitPartial)->Arg(1)->Arg(4096)->Arg(16384)->Arg(65536)
 static void BufferLinearizeSimple(benchmark::State& state) {
   const std::string data(state.range(0), 'a');
   const absl::string_view input(data);
-  Buffer::BufferFragmentImpl fragment(input.data(), input.size(), DoNotReleaseFragment);
   Buffer::OwnedImpl buffer;
   for (auto _ : state) {
     buffer.drain(buffer.length());
-    buffer.addBufferFragment(fragment);
+    auto fragment =
+        std::make_unique<Buffer::BufferFragmentImpl>(input.data(), input.size(), DeleteFragment);
+    buffer.addBufferFragment(*fragment.release());
```

Comment:
Why make this smart pointer instead of  just using `new` to get the raw pointer?

Reply:
There's a similar pattern used in source/common/http/http1/codec_impl.cc and source/common/http/http2/codec_impl.cc

I think the pattern is a bit more robust against memory leaks in paths that end up throwing.  There's some problems with the addBufferFragment API, perhaps it should be changed to taking an unique_ptr and remove the need to call a done method that acts like a destructor.  But that's outside the scope of this PR.

Reply:
Technically I think because you call release right away there is no functional difference if you just inline the new directly in the function call, but it's definitely safer if it remains a local variable. I think it's fine either way so let's merge this.